### PR TITLE
Using ServiceMonitors directly

### PR DESCRIPTION
**Pull request by dgiebert**
## Description

Changes [Metrics Quickstart](https://docs.kubewarden.io/operator-manual/telemetry/metrics/quickstart) to use ServiceMonitors instead of modifying the Prometheus Chart

### Tradeoff

The Kubewarden UI needs to be adapted to search for ServiceMonitors and not the Chart

### CODE REVIEW

File: versioned_docs/version-1.9/operator-manual/telemetry/metrics/01-quickstart.md [hunk 1]
```diff
@@ -19,72 +19,69 @@ Server.
 ## Install Prometheus
 
 We will use the [Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator),
-that allows us to intuitively define Prometheus' Targets.
+that allows us to define Prometheus' Targets intuitively.
 
 There are many ways to install and set up Prometheus. For ease of deployment, we will use the
-Prometheus community Helm chart.
+Prometheus community Helm chart, but you can also use the Rancher Monitoring Chart.
```
[the remark this answers is not in the document]
Comment (dgiebert) [reply versioned_docs/version-1.9/operator-manual/telemetry/metrics/01-quickstart.md:25]: Oh! Yes I will remove it.

**Comment by dgiebert** [comment]
I clicked edit on the main page, but now really checking the source shouldn't this go in the unversioned section of the docs?

**Comment by jordojordo** [comment]
Could you also remove the references to the Prometheus spec within the UI docs? https://docs.kubewarden.io/operator-manual/ui-extension/metrics#1-within-the-cluster-explorer-click-on-cluster-tools-in-the-side-navigation

**Review by viccuad — CHANGES_REQUESTED** [review]
> I clicked edit on the main page, but now really checking the source shouldn't this go in the unversioned section of the docs?

Argh, yes, you are right, sorry. This should go into the `docs/` folder.

**Comment by dgiebert** [comment]
> Could you also remove the references to the Prometheus spec within the UI docs? https://docs.kubewarden.io/operator-manual/ui-extension/metrics#1-within-the-cluster-explorer-click-on-cluster-tools-in-the-side-navigation

Can do, but shouldn't there be a pointer to the Install Section for Prometheus/Opentelemetry?

**Comment by jordojordo** [comment]
> shouldn't there be a pointer to the Install Section for Prometheus/Opentelemetry?
Yes you're right, there should be at least a pointer to the section.

**Comment by dgiebert** [comment]
@jordojordo adapted the UI section of the documentation as well

**Review by jordojordo — APPROVED** [review]
LGTM

**Review by kravciak — CHANGES_REQUESTED** [review]
namespaceSelectors in UI doc point to `kubewarden`, but we use `cattle-kubewarden-system` in Rancher

**Review by kravciak — CHANGES_REQUESTED** [review]

**Comment by kravciak** [comment]
> > namespaceSelectors point to `kubewarden`, but we use `cattle-kubewarden-system` in Rancher
> 
> I can change that, but all docs use the `kubewarden` namespace (e.g. [here](https://docs.kubewarden.io/operator-manual/telemetry/metrics/quickstart#install-kubewarden))

Section of [doc about UI extension](https://docs.kubewarden.io/operator-manual/ui-extension/install) use cattle-kubewarden-system ns. Kubewarden namespace does not exist if you install through extension.

File: docs/operator-manual/ui-extension/metrics.md [hunk 2]
```diff
@@ -12,35 +12,50 @@ You will need a cluster with at least 4 cores to install the Monitoring tool.
...
+metadata:
+  name: kubewarden-controller
+  namespace: kubewarden
+spec:
+  endpoints:
+    - interval: 10s 
+      port: metrics
+  namespaceSelector:
+    matchNames:
+      - kubewarden
```
Comment (viccuad) [inline docs/operator-manual/ui-extension/metrics.md:39]:
```suggestion
      - cattle-kubewarden-system
```

File: docs/operator-manual/ui-extension/metrics.md [hunk 3]
```diff
@@ -12,35 +12,50 @@ You will need a cluster with at least 4 cores to install the Monitoring tool.
...
+metadata:
+  name: kubewarden-policy-server
+  namespace: kubewarden
+spec:
+  endpoints:
+    - interval: 10s
+      port: metrics
+  namespaceSelector:
+    matchNames:
+      - kubewarden
```
Comment (viccuad) [inline docs/operator-manual/ui-extension/metrics.md:55]:
```suggestion
      - cattle-kubewarden-system
```

**Comment by dgiebert** [comment]
Ok, will fix and implement the changes this evening

File: docs/operator-manual/ui-extension/metrics.md [hunk 4]
```diff
@@ -12,35 +12,50 @@ You will need a cluster with at least 4 cores to install the Monitoring tool.
...
-      endpoints:
-        - port: metrics
-          interval: 10s
-  annotations: {}
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: kubewarden-controller
+  namespace: kubewarden
```
Comment (kravciak) [inline docs/operator-manual/ui-extension/metrics.md:32]:
```suggestion
  namespace: cattle-kubewarden-system
```

File: docs/operator-manual/ui-extension/metrics.md [hunk 5]
```diff
@@ -12,35 +12,50 @@ You will need a cluster with at least 4 cores to install the Monitoring tool.
...
+      - kubewarden
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: kubewarden-controller
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: kubewarden-policy-server
+  namespace: kubewarden
```
Comment (kravciak) [inline docs/operator-manual/ui-extension/metrics.md:48]:
```suggestion
  namespace: cattle-kubewarden-system
```

**Comment by dgiebert** [comment]
Done, also checked with `grep -r 'namespace: kubewarden'` there is no other mention in the docs :+1:

File: docs/operator-manual/telemetry/metrics/01-quickstart.md [hunk 6]
```diff
@@ -19,72 +19,69 @@ Server.
...
+  prometheus prometheus-community/kube-prometheus-stack
 ```
 
 The `prometheus-operator` deployed as part of this Helm chart defines the concept of [Service
 Monitors](https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/design.md#servicemonitor),
-used to declaratively define which services should be monitored by Prometheus.
+ to define which services should be monitored by Prometheus declaratively.
 
-In our case, we are adding a Service monitor targeting the `kubewarden` namespace, for services that
+In our case, we are adding a Service monitor targeting the `kubewarden` namespace for services that
```
Comment (viccuad) [inline docs/operator-manual/telemetry/metrics/01-quickstart.md:47]:
```suggestion
In our case, we are adding a ServiceMonitor targeting the `kubewarden` namespace for services that
```

File: docs/operator-manual/ui-extension/metrics.md [hunk 7]
```diff
@@ -12,35 +12,50 @@ You will need a cluster with at least 4 cores to install the Monitoring tool.
 
 ## Prerequisites
 
-Cert-Manager and OpenTelemetry are required.
-Follow [these instructions](../telemetry/opentelemetry/01-quickstart.md#install-opentelemetry) to install Cert Manager and the OpenTelemetry Operator.
+The Prometheus Operator is required.
```
[the remark this answers is not in the document]
Comment (dgiebert) [reply docs/operator-manual/ui-extension/metrics.md:15]: Could change this to say "A Prometheus Operator is required" that would also include the Rancher one as an option?

File: docs/operator-manual/telemetry/metrics/01-quickstart.md [hunk 8]
```diff
@@ -19,72 +19,69 @@ Server.
...
+      - kubewarden
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: kubewarden-controller
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: kubewarden-policy-server
+  namespace: cattle-kubewarden-system
```
Comment (viccuad) [inline docs/operator-manual/telemetry/metrics/01-quickstart.md:74]:
```suggestion
  namespace: kubewarden
```

File: docs/operator-manual/ui-extension/metrics.md [hunk 9]
```diff
@@ -12,35 +12,50 @@ You will need a cluster with at least 4 cores to install the Monitoring tool.
...
+metadata:
+  name: kubewarden-controller
+  namespace: cattle-kubewarden-system
+spec:
+  endpoints:
+    - interval: 10s 
+      port: metrics
+  namespaceSelector:
+    matchNames:
+      - kubewarden
```
Comment (viccuad) [inline docs/operator-manual/ui-extension/metrics.md:39]:
```suggestion
      - cattle-kubewarden-system
```

File: docs/operator-manual/ui-extension/metrics.md [hunk 10]
```diff
@@ -12,35 +12,50 @@ You will need a cluster with at least 4 cores to install the Monitoring tool.
...
+metadata:
+  name: kubewarden-policy-server
+  namespace: cattle-kubewarden-system
+spec:
+  endpoints:
+    - interval: 10s
+      port: metrics
+  namespaceSelector:
+    matchNames:
+      - kubewarden
```
Comment (viccuad) [inline docs/operator-manual/ui-extension/metrics.md:55]:
```suggestion
      - cattle-kubewarden-system
```

File: docs/operator-manual/telemetry/metrics/01-quickstart.md [hunk 11]
```diff
@@ -19,72 +19,69 @@ Server.
...
-  --namespace prometheus \
-  --version 51.5.3 \
-  --values kube-prometheus-stack-values.yaml \
-  prometheus prometheus-community/kube-prometheus-stack
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: kubewarden-controller
+  namespace: cattle-kubewarden-system
```
[the remark this answers is not in the document]
Comment (dgiebert) [reply docs/operator-manual/telemetry/metrics/01-quickstart.md:58]: Don't we install everything then to `cattle-kubewarden-system`?

**Review by viccuad — APPROVED** [review]
Many thanks and sorry for the confusion!

**Review by kravciak — APPROVED** [review]

**Comment by kravciak** [comment]
Thanks for the contribution!

**Pull request merged**